### ISSUE TITLE
a11y: Narrator on Win 11 reads the message in SemanticsService.announce() twice

### ISSUE THREAD

**Issue by fdennis** [flutter/flutter#153020]
### Steps to reproduce

Use `SemanticsService.announce()` on Win 11 and verify with a screen reader, such as Narrator, that the message is read.

I have attached a code sample where I have added `SemanticsService.announce("test", TextDirection.ltr);` to the `_incrementCounter()` function. Whenever the `increment` button is pressed, the text "test" should be read once when a screen reader is active.

### Expected results

The message in `announce()` is read once. 

### Actual results

The message in `announce()` is read twice.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // TRY THIS: Try running your application with "flutter run". You'll see
        // the application has a purple toolbar. Then, without quitting the app,
        // try changing the seedColor in the colorScheme below to Colors.green
        // and then invoke "hot reload" (save your changes or press the "hot
        // reload" button in a Flutter-supported IDE, or press "r" if you used
        // the command line to start the app).
        //
        // Notice that the counter didn't reset back to zero; the application
        // state is not lost during the reload. To reset the state, use hot
        // restart instead.
        //
        // This works for code too, not just values: Most code changes can be
        // tested with just a hot reload.
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    SemanticsService.announce("test", TextDirection.ltr);
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // TRY THIS: Try changing the color here to a specific color (to
        // Colors.amber, perhaps?) and trigger a hot reload to see the AppBar
        // change color while the other colors stay the same.
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          //
          // TRY THIS: Invoke "debug painting" (choose the "Toggle Debug Paint"
          // action in the IDE, or press "p" in the console), to see the
          // wireframe for each widget.
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by huycozy** [comment]
Hi @fdennis, I checked this on Windows 10 and heard the reader only read one time as: `alert test`. I'm on the latest Flutter stable version 3.24.0. I haven't checked on Windows 11 yet but will do if necessary.

What is your Flutter version? And which platform are you checking on? (Windows or Web?)

**Comment by huycozy** [comment]
Hi @fdennis, I checked this on Windows 10 and heard the reader only read one time as: `alert test`. I'm on the latest Flutter stable version 3.24.0. I haven't checked on Windows 11 yet but will do if necessary.

What is your Flutter version? Please provide the output of `flutter doctor -v`. 

And which platform are you checking on? (Windows or Web?)

**Comment by fdennis** [comment]
Hi @huycozy 
I am on the latest stable version as well, 3.24.0 and I hear `alert test alert test` from my code sample.
The platform is Windows.

I am not sure what exact parts of the `flutter doctor -v` output you are interested in, but here it is (I have removed some parts):
```
[✓] Flutter (Channel stable, 3.24.0, on Microsoft Windows [Version
    10.0.22631.3593], locale en-US)
    • Flutter version 3.24.0 on channel stable at
      C:\PATH\flutter_windows_3.19.5-stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (9 days ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✗] Android toolchain - develop for Android devices
[REMOVED]

[✓] Chrome - develop for the web
[REMOVED]

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34622.214
    • Windows 10 SDK version 10.0.22621.0

[!] Android Studio (not installed)
[REMOVED]

[✓] Connected device (3 available)
[REMOVED]

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

**Comment by huycozy** [comment]
I can reproduce this issue on Windows 11. The results are not the same as below in variety readers:

- Narrator: `alert test`, `alert test` (read twice)
- NVDA: `test alert` (read one, seems to correct)
- Jaws: `test test` (read twice)

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[!] Flutter (Channel stable, 3.24.0, on Microsoft Windows [Version 10.0.22631.3296], locale en-US)
    • Flutter version 3.24.0 on channel stable at D:\Working\GitHub\flutter
    ! Warning: `flutter` on your path resolves to D:\Working\GitHub\Aliases\flutter.bat, which is not inside your current Flutter SDK checkout at
      D:\Working\GitHub\flutter. Consider adding D:\Working\GitHub\flutter\bin to the front of your path.
    ! The dart binary is not on your path. Consider adding D:\Working\GitHub\flutter\bin to your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (9 days ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and
      upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at D:\android-sdk-windows
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34622.214
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2023.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[√] VS Code (version 1.88.0)
    • VS Code at C:\Users\Huy\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.84.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3296]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 127.0.6533.90
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 123.0.2420.81

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.24.0-1.0.pre.509, on Microsoft Windows [Version 10.0.22631.3296], locale en-US)
    • Flutter version 3.24.0-1.0.pre.509 on channel master at D:\Working\GitHub\flutter_master
    ! Warning: `flutter` on your path resolves to D:\Working\GitHub\Aliases\flutter.bat, which is not inside your current Flutter SDK checkout at
      D:\Working\GitHub\flutter_master. Consider adding D:\Working\GitHub\flutter_master\bin to the front of your path.
    ! The dart binary is not on your path. Consider adding D:\Working\GitHub\flutter_master\bin to your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 925ad4e6fa (7 hours ago), 2024-08-08 00:38:29 -0400
    • Engine revision 94cf8c8fad
    • Dart version 3.6.0 (build 3.6.0-127.0.dev)
    • DevTools version 2.38.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and
      upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at D:\android-sdk-windows
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34622.214
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2023.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[√] VS Code (version 1.88.0)
    • VS Code at C:\Users\Huy\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.86.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3296]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 127.0.6533.90
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 123.0.2420.81

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>
